### PR TITLE
Remediating some warnings when using KWIVER_CPP_EXTRA option.

### PR DESCRIPTION
**Pull request by linus-sherrill**
Specifically addressing 
- unused parameters
- shadowed variables
- Missing case elements of a switch

### CODE REVIEW

File: vital/algo/bundle_adjust.cxx [hunk 1]
```diff
@@ -62,8 +62,8 @@ ::optimize(
   kwiver::vital::simple_camera_perspective_map &cameras,
   kwiver::vital::landmark_map::map_landmark_t &landmarks,
   vital::feature_track_set_sptr tracks,
-  const std::set<vital::frame_id_t>& fixed_cameras,
-  const std::set<vital::landmark_id_t>& fixed_landmarks,
+  const std::set<vital::frame_id_t>& /* fixed_cameras */,
```
Comment (dstoup) [inline vital/algo/bundle_adjust.cxx:65]: Why use VITAL_UNUSED in some places and commenting out in others?

**Review by dstoup — COMMENTED** [review]
LGTM

File: vital/algo/bundle_adjust.cxx (hunk 1 above)
Comment (linus-sherrill) [reply vital/algo/bundle_adjust.cxx]: Don't know why this was handled as a comment. Has been updated to use VITAL_UNUSED

**Comment by linus-sherrill** [comment]
If this cleans up some classes of warnings, we can move some flags, like warnings for unused parameters and shadowing to the main set of options.

**Comment by dstoup** [comment]
> If this cleans up some classes of warnings, we can move some flags, like warnings for unused parameters and shadowing to the main set of options.

I agree! I would be great to move more flags into the main set of warnings. Probably good to do it even if they aren't completely cleaned up with this PR.  Once they show up in the builds we can clean the rest up petty quickly.

**Pull request closed without merging**